doc.go

```diff
@@ -1,4 +1,190 @@
+
+# Query cancellation and timeout
+
+Cancelling a query via context cancellation or timeout is supported.
+
+	// Set up context timeout
+	ctx, cancel := context.WithTimeout(context.Background(), 30 * time.Second)
+	defer cancel()
+
+	// Execute query. Query will be cancelled after 30 seconds if still running
+	res, err := db.ExecContext(ctx, "CREATE TABLE example(id int, message string)")
+
+# CorrelationId and ConnId
+
+Use the driverctx package under driverctx/ctx.go to add CorrelationId and ConnId to the context.
+CorrelationId and ConnId makes it convenient to parse and create metrics in logging.
+
+**Connection Id**
+Internal id to track what happens under a connection. Connections can be reused so this would track across queries.
+
+**Query Id**
+Internal id to track what happens under a query. Useful because the same query can be used with multiple connections.
+
+**Correlation Id**
+External id, such as request ID, to track what happens under a request. Useful to track multiple connections in the same request.
+
+	ctx := dbsqlctx.NewContextWithCorrelationId(context.Background(), "workflow-example")
+
+# Logging
+
+Use the logger package under logger.go to set up logging (from zerolog).
+By default, logging level is `warn`. If you want to disable logging, use `disabled`.
+The user can also utilize Track() and Duration() to custom log the elapsed time of anything tracked.
+
+	import (
+		dbsqllog "github.com/databricks/databricks-sql-go/logger"
+		dbsqlctx "github.com/databricks/databricks-sql-go/driverctx"
+	)
+
+	func main() {
+		// Optional. Set the logging level with SetLogLevel()
+		if err := dbsqllog.SetLogLevel("debug"); err != nil {
+			log.Fatal(err)
+		}
+
+		// Optional. Set logging output with SetLogOutput()
+		// Default is os.Stderr. If running in terminal, logger will use ConsoleWriter to prettify logs
+		dbsqllog.SetLogOutput(os.Stdout)
+
+		// Optional. Set correlation id with NewContextWithCorrelationId
+		ctx := dbsqlctx.NewContextWithCorrelationId(context.Background(), "workflow-example")
+
+
+		// Optional. Track time spent and log elapsed time
+		msg, start := logger.Track("Run Main")
+		defer log.Duration(msg, start)
+
+		db, err := sql.Open("databricks", "<dsn_string>")
+		...
+	}
+
+The result log may look like this:
+
+	{"level":"debug","connId":"01ed6545-5669-1ec7-8c7e-6d8a1ea0ab16","corrId":"workflow-example","queryId":"01ed6545-57cc-188a-bfc5-d9c0eaf8e189","time":1668558402,"message":"Run Main elapsed time: 1.298712292s"}
+
+# Supported Data Types
+
+==================================
+
+Databricks Type --> Golang Type
+
+==================================
+
+BOOLEAN --> bool
+
+TINYINT --> int8
+
+SMALLINT --> int16
+
+INT --> int32
+
+BIGINT --> int64
+
+FLOAT --> float32
+
+DOUBLE --> float64
+
+VOID --> nil
+
+STRING --> string
+
+DATE --> time.Time
+
+TIMESTAMP --> time.Time
+
+DECIMAL(p,s) --> sql.RawBytes
+
+BINARY --> sql.RawBytes
+
+ARRAY<elementType> --> sql.RawBytes
```

Comment:
maybe worth mentioning that ARRAY, STRUCT, MAP can be scanned to a string, in which will be like:
`{arrayVal:[1,2,3] mapVal:{"key1":1} structVal:{"string_field":"string_val","array_field":[4,5,6]}}`
And from there the user can unmarshal the JSON string to the appropriate type